openmls/src/group/tests/test_external_commit_validation.rs

```diff
@@ -0,0 +1,886 @@
+        mut plaintext,
+        original_plaintext,
+    } = validation_test_setup(*PURE_PLAINTEXT_WIRE_FORMAT_POLICY, ciphersuite, backend);
+
+    let mut content = if let MlsPlaintextContentType::Commit(commit) = plaintext.content() {
+        commit.clone()
+    } else {
+        panic!("Unexpected content type.");
+    };
+
+    // Insert a second external init proposal into the commit.
+    let second_ext_init_prop =
+        ProposalOrRef::Proposal(Proposal::ExternalInit(ExternalInitProposal::from(vec![
+            1, 2, 3,
+        ])));
+
+    content.proposals.push(second_ext_init_prop);
+
+    plaintext.set_content(MlsPlaintextContentType::Commit(content));
+
+    // We have to re-sign, since we changed the content.
+    let mut signed_plaintext: MlsPlaintext = plaintext
+        .payload()
+        .clone()
+        .sign(backend, &bob_credential_bundle)
+        .expect("Error signing modified payload.");
+
+    // Set old confirmation tag
+    signed_plaintext.set_confirmation_tag(
+        original_plaintext
+            .confirmation_tag()
+            .expect("no confirmation tag on original message")
+            .clone(),
+    );
+
+    let verifiable_plaintext: VerifiableMlsPlaintext =
+        VerifiableMlsPlaintext::from_plaintext(signed_plaintext, None);
+
+    // Have alice process the commit resulting from external init.
+    let message_in = MlsMessageIn::from(verifiable_plaintext);
+
+    let unverified_message = alice_group
+        .parse_message(message_in, backend)
+        .expect("Could not parse message.");
+
+    let err = alice_group
+        .process_unverified_message(unverified_message, None, backend)
+        .expect_err(
+            "Could process unverified message despite second ext. init proposal in commit.",
+        );
+
+    assert_eq!(
+        err,
+        MlsGroupError::Group(CoreGroupError::ExternalCommitValidationError(
+            ExternalCommitValidationError::MultipleExternalInitProposals
+        ))
+    );
+
+    // Positive case
+    let unverified_message = alice_group
+        .parse_message(MlsMessageIn::from(original_plaintext), backend)
+        .expect("Could not parse message.");
+    alice_group
+        .process_unverified_message(unverified_message, None, backend)
+        .expect("Unexpected error.");
+}
+
+// ValSem242: External Commit, inline Proposals: There MUST NOT be any Add proposals.
+#[apply(ciphersuites_and_backends)]
+fn test_valsem242(ciphersuite: &'static Ciphersuite, backend: &impl OpenMlsCryptoProvider) {
+    // Test with MlsPlaintext
+    let ECValidationTestSetup {
+        mut alice_group,
+        bob_credential_bundle,
+        mut plaintext,
+        original_plaintext,
+    } = validation_test_setup(*PURE_PLAINTEXT_WIRE_FORMAT_POLICY, ciphersuite, backend);
+
+    let mut content = if let MlsPlaintextContentType::Commit(commit) = plaintext.content() {
+        commit.clone()
+    } else {
+        panic!("Unexpected content type.");
```

Comment:
I'm still wondering how we can convince codecov that the `panic` is valid here and doesn't need additional testing. I don't have a suggestion though and this doesn't need to be changed.